File: backend/app/api/health.py
"""健康检查和安全配置诊断接口。"""

from datetime import UTC, datetime

from fastapi import APIRouter
from pydantic import BaseModel

from app.config import get_settings

router = APIRouter(tags=["health"])


class LlmConfigStatus(BaseModel):
    """可安全返回给前端的大模型配置状态。"""

    provider: str
    base_url: str
    model: str
    api_key_configured: bool
    api_key_preview: str
    timeout_seconds: int


class HealthResponse(BaseModel):
    """健康检查响应体。"""

    status: str
    app_name: str
    app_version: str
    app_env: str
    checked_at: datetime
    llm: LlmConfigStatus


@router.get("/health", response_model=HealthResponse)
def health_check() -> HealthResponse:
    """返回服务健康状态和脱敏后的大模型配置。"""
    settings = get_settings()
    return HealthResponse(
        status="ok",
        app_name=settings.app_name,
        app_version=settings.app_version,
        app_env=settings.app_env,
        checked_at=datetime.now(UTC),
        llm=LlmConfigStatus(
            provider=settings.llm_provider,
            base_url=settings.llm_base_url,
            model=settings.llm_model,
            api_key_configured=settings.llm_api_key_configured,
            api_key_preview=settings.masked_llm_api_key,
            timeout_seconds=settings.llm_timeout_seconds,
        ),
    )

File: frontend/src/types/health.ts
/** 大模型配置的脱敏健康状态。 */
export interface LlmConfigStatus {
  provider: string; // provider 名称
  base_url: string; // 中转站基础地址
  model: string; // 当前模型名
  api_key_configured: boolean; // 是否已配置密钥
  api_key_preview: string; // 脱敏后的密钥预览
  timeout_seconds: number; // 请求超时时间
  status?: string; // 前端可展示的简短状态
}

/** 更新大模型运行时配置时提交的请求体。 */
export interface UpdateLlmConfigPayload {
  base_url: string; // OpenAI-compatible API 基础地址
  model: string; // 模型名称
  api_key?: string; // API Key；留空时后端保留当前密钥
}

/** 健康检查接口响应。 */
export interface HealthResponse {
  status: string; // 服务状态
  app_name: string; // 应用名
  app_version: string; // 应用版本
  app_env: string; // 运行环境
  checked_at: string; // 检查时间
  llm: LlmConfigStatus; // 大模型配置状态
}
